Neos.Flow/Classes/Mvc/Routing/UriBuilder.php

```diff
@@ -361,7 +361,7 @@ public function build(array $arguments = [])
         $uriPathPrefix = RequestInformationHelper::getScriptRequestPath($httpRequest) . $uriPathPrefix;
         $uriPathPrefix = ltrim($uriPathPrefix, '/');
 
-        $resolveContext = new ResolveContext($this->baseUriProvider->getConfiguredBaseUriOrFallbackToCurrentRequest(), $arguments, $this->createAbsoluteUri, $uriPathPrefix);
+        $resolveContext = new ResolveContext($this->baseUriProvider->getConfiguredBaseUriOrFallbackToCurrentRequest($httpRequest), $arguments, $this->createAbsoluteUri, $uriPathPrefix);
```

Comment:
Aren't we running in circles with this? As in the `$httpRequest` in CLI here is just the fake http request that got made up to create the ActionRequest to inject into the UriBuilder? So what is the URI of that fake http request? And can't we simply use that here directly?

Reply:
That's what I also suggested when I talked to @kitsunet. But this would change the behavior as currently the configured baseUri has precedence over the current request URI.

Or, to put it simple:
When you have a `baseUri` configured, links will have that URI prefixed if you come from a different domain.